File: Source/Aladdin.ASN1.ISO/ASN1/ISO/OCSP/PreferredSignatureAlgorithm.cs
using System;
using System.Runtime.Serialization;

// PreferredSignatureAlgorithm ::= SEQUENCE {
//    sigIdentifier  AlgorithmIdentifier{SIGNATURE-ALGORITHM, {...}},
//    certIdentifier AlgorithmIdentifier{PUBLIC-KEY, {...}} OPTIONAL
// }

namespace Aladdin.ASN1.ISO.OCSP
{
	[Serializable]
	public class PreferredSignatureAlgorithm : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<AlgorithmIdentifier>().Factory(), Cast.O) 
		}; 
		// конструктор при сериализации
        protected PreferredSignatureAlgorithm(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public PreferredSignatureAlgorithm(IEncodable encodable) : base(encodable, info) {} 

		// конструктор при закодировании
		public PreferredSignatureAlgorithm(AlgorithmIdentifier sigIdentifier, AlgorithmIdentifier certIdentifier) : 
			base(info, sigIdentifier, certIdentifier) {} 

		public AlgorithmIdentifier	SigIdentifier	{ get { return (AlgorithmIdentifier	)this[0]; } } 
		public AlgorithmIdentifier	CertIdentifier	{ get { return (AlgorithmIdentifier	)this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/ocsp/PreferredSignatureAlgorithm.java
package aladdin.asn1.iso.ocsp;
import aladdin.asn1.*;
import aladdin.asn1.iso.*;
import java.io.*;

// PreferredSignatureAlgorithm ::= SEQUENCE {
//    sigIdentifier  AlgorithmIdentifier{SIGNATURE-ALGORITHM, {...}},
//    certIdentifier AlgorithmIdentifier{PUBLIC-KEY, {...}} OPTIONAL
// }

public class PreferredSignatureAlgorithm extends Sequence<IEncodable>
{
    private static final long serialVersionUID = 4422999841875621444L;

    // информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(AlgorithmIdentifier.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public PreferredSignatureAlgorithm(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public PreferredSignatureAlgorithm(AlgorithmIdentifier sigIdentifier, AlgorithmIdentifier certIdentifier) 
	{ 
		super(info, sigIdentifier, certIdentifier); 
	}
	public final AlgorithmIdentifier  sigIdentifier () { return (AlgorithmIdentifier)get(0); } 
	public final AlgorithmIdentifier  certIdentifier() { return (AlgorithmIdentifier)get(1); }
}
